main.py

```diff
@@ -44,17 +50,58 @@ def checkout_commit(commit_hash, directory):
         print(f"Failed to checkout commit {commit_hash} in {directory}")
         
 
-def run_specimin(build_command):
-    pass
+def build_specimin_command(issue_id, root_dir, package_name, targets):
```

Comment:
`issue_id` here does not actually have to be an issue ID, does it? I think it's actually the name to use for the subdirectory in which this command will be executed. Maybe `issue_id` is fine as a name (since that's what it will be in practice), but we should document that there's no particular required format.

By contrast, I think `package_name` does need to be a valid Java package name.

Reply:
This leaves me a bit unsure of the purpose of `root_dir`, though. It seems like the `root_dir` variable is reused on line 55, and that this parameter is part of that path? But why? I don't see the reason that we need a separate variable for this (`issue_id` should be enough to differentiate each test, shouldn't it?).

Reply:
You are right Professor.  issue_id is the name of the folder that contains the specimin's target project. Few examples of issue_id are cf-1291, cf-3032, jdk-8269121 etc. Each issue_id folder contains the specimin's target project in the input folder and specimin's result in the output folder. The folder structure is like below:

ISSUES
│
|── cf-1291
│   |── input   --> contains the project on which specimin is executed
|   |--------|nomulus/core/src/main/java/.    --> this is the root dir
|   |---------------------------------|package_path/file.java
│   │── output --> contains the minimized project after specimin is executed 
│   │   

the args for the gradle command would be `'--root "nomulus/core/src/main/java/" --targetFile "package_path/file.java"... '`

Since the gradle command of specimin uses fully qualified path of a target file relative to its root directory which is not necessarily the base directory of the project, I am using the root directory here.

Reply:
I see, thanks for the explanation. Please add a comment to the code that explains this directory structure so that we don't have to reverse-engineer it in the future.